subsys/logging/log_core.c

```diff
@@ -45,7 +45,7 @@ LOG_MODULE_REGISTER(log);
 
 struct log_strdup_buf {
 	atomic_t refcount;
-	char buf[CONFIG_LOG_STRDUP_MAX_STRING + 1]; /* for termination */
+	char buf[CONFIG_LOG_STRDUP_MAX_STRING + 2]; /* for termination */
```

Comment:
Should probably update the comment to explain why this is 2.

Reply:
Well, it's still "for termination" but I can see what you mean.  Would "for ~\0 termination" be clearer or did you have something else in mind?

Reply:
I think `for ~\0 terminator string` works.